### PR TITLE
[NavigationBar] Add titleView tests.

### PR DESCRIPTION
**Pull request by jverkoey**
These tests will help validate the fix for https://github.com/material-components/material-components-ios/issues/7207

### CODE REVIEW

File: components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  func testNavigationItemTitleViewAssignmentWithSimulatedTheftKeepsTitleViewAsSubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+    let navigationItem = UINavigationItem()
+
+    // When
+    navigationItem.titleView = titleView
+    navigationBar.observe(navigationItem)
+    let simulatedThiefView = UIView()
```
Comment (yarneo) [inline components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift]: should this instantiation be moved to Given?

File: components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    simulatedThiefView.addSubview(navigationItem.titleView!)
+
+    // Then
+    XCTAssertEqual(navigationItem.titleView?.superview, simulatedThiefView)
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  // Designed to keep https://github.com/material-components/material-components-ios/issues/7207
```
Comment (yarneo) [inline components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift:123]: is this expected to pass after the fix?

**Review by yarneo — APPROVED** [review]

File: components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift (hunk 2 above)
Comment (jverkoey) [reply components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift:123]: Yep.

File: components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift (hunk 1 above)
Comment (jverkoey) [reply components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift]: Done.

**Pull request merged**